test/AquaTest.jl

```diff
@@ -0,0 +1,29 @@
+module AquaTest
+
+using Test
+using ForwardDiff
+using Aqua
+
+@testset "Aqua tests - unbound_args" begin
+    # This tests that we don't accidentally run into
+    # https://github.com/JuliaLang/julia/issues/29393
+    ua = Aqua.detect_unbound_args_recursively(ForwardDiff)
+    @test length(ua) == 6
+end
```

Comment:
Ideally the unbound arguments should be fixed, which would also allow us to just run the default `Aqua.test_all` test with `unbound_args = true` below instead of this custom test.

Reply:
These ambiguities seem to be upstream issues. See https://github.com/JuliaTesting/Aqua.jl/issues/86 and https://github.com/JuliaLang/julia/issues/28086.

We can "fix" this (I'm not 100% sure if the issue is here or upstream, even based on discussions) by changing e.g.,

```julia
@generated function one_tuple(::Type{NTuple{N,V}}) where {N, V}
```
to
```julia
@generated function one_tuple(::Type{Tuple{V,Vararg{R1,N}}) where {N, V}
```
Should I move forward with those changes?